From 324dcd373371fbc2855406444c1c106722970c18 Mon Sep 17 00:00:00 2001
From: Brian Behlendorf <behlendorf1@llnl.gov>
Date: Tue, 1 Sep 2015 09:23:02 -0700
Subject: [PATCH] Linux 4.2 compat: misc_deregister()

The misc_deregister() function was changed to a void return type.
Rather than add compatibility code to detect this change simply
ignore the return code on all kernels.  It was only used to log
an informational error message of no real value.

Signed-off-by: Brian Behlendorf <behlendorf1@llnl.gov>
---
 module/zfs/zfs_ioctl.c | 6 +-----
 module/zpios/pios.c    | 6 +-----
 2 files changed, 2 insertions(+), 10 deletions(-)

diff --git a/module/zfs/zfs_ioctl.c b/module/zfs/zfs_ioctl.c
index ba695dd..61bc4b8 100644
--- a/module/zfs/zfs_ioctl.c
+++ b/module/zfs/zfs_ioctl.c
@@ -6028,13 +6028,9 @@ zfs_attach(void)
 static void
 zfs_detach(void)
 {
-	int error;
 	zfsdev_state_t *zs, *zsprev = NULL;
 
-	error = misc_deregister(&zfs_misc);
-	if (error != 0)
-		printk(KERN_INFO "ZFS: misc_deregister() failed %d\n", error);
-
+	misc_deregister(&zfs_misc);
 	mutex_destroy(&zfsdev_state_lock);
 
 	for (zs = zfsdev_state_list; zs != NULL; zs = zs->zs_next) {
diff --git a/module/zpios/pios.c b/module/zpios/pios.c
index 8e5077d..e3a85c1 100644
--- a/module/zpios/pios.c
+++ b/module/zpios/pios.c
@@ -1310,11 +1310,7 @@ zpios_init(void)
 static void __exit
 zpios_fini(void)
 {
-	int error;
-
-	error = misc_deregister(&zpios_misc);
-	if (error)
-		printk(KERN_INFO "ZPIOS: misc_deregister() failed %d\n", error);
+	misc_deregister(&zpios_misc);
 
 	printk(KERN_INFO "ZPIOS: Unloaded module v%s-%s%s\n",
 	    ZFS_META_VERSION, ZFS_META_RELEASE, ZFS_DEBUG_STR);
-- 
2.5.0

